syslog-ng-rs/syslog-ng-common/src/proxies/parser/mod.rs

```diff
@@ -54,58 +56,123 @@ pub mod _parser_plugin {
 
     #[no_mangle]
     pub extern fn native_parser_proxy_init(this: &mut ParserProxy<$name>) -> c_int {
-        let result = this.init();
-        bool_to_int(result)
+        let mut wrapper = AssertUnwindSafe(this);
+
+        match catch_unwind(move || wrapper.init()) {
+            Ok(init_result) => bool_to_int(init_result),
+            Err(error) => {
+                error!("native_parser_proxy_init() panicked, but the panic was caught: {:?}", error);
+                commit_suicide();
+            }
+        }
     }
 
     #[no_mangle]
     pub extern fn native_parser_proxy_deinit(this: &mut ParserProxy<$name>) -> c_int {
-        let result = this.deinit();
-        bool_to_int(result)
+        let mut wrapper = AssertUnwindSafe(this);
+
+        match catch_unwind(move || wrapper.deinit()) {
+            Ok(value) => bool_to_int(value),
```

Comment:
I checked the names and tried to use descriptive ones, like `process_result`, `cloned` etc.